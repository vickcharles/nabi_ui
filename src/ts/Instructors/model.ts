export interface ExperienceType {
  id?: number;
  employer: string;
  jobTitle: string;
  fromMonth: string;
  fromYear: string;
  toMonth: string;
  toYear: string;
  location: string;
}

export enum DegreeType {
  associate = 'Associate Degree',
  bachelors = 'Bachelor\'s Degree',
  graduate = 'Graduate Degreee',
  professional = 'Professional Degree',
  certification = 'Certification',
  other = 'Other'
}

export interface EducationType {
  id?: number;
  schoolName: string;
  graduationYear: string;
  degreeType: DegreeType;
  fieldOfStudy: string;
}

export enum PlaceForLessonsOptions {
  home = 'Student\'s home',
  studio = 'Instructor\'s studio',
  online = 'Online',
}

export interface PlaceForLessonsType {
  home: boolean;
  studio: boolean;
  online: boolean;
}

export enum LessonDuration {
  thirtyMins = '30 mins',
  fortyFiveMins = '45 mins',
  sixtyMins = '60 mins',
  ninetyMins = '90 mins'
}

export interface RatesType {
  thirtyMinsRate: number;
  fortyFiveMinsRate: number;
  sixtyMinsRate: number;
  ninetyMinsRate: number;
}

export enum SkillLevel {
  beginner = 'Beginner',
  intermediate = 'Intermediate',
  advanced = 'Advanced'
}

export interface InstrumentsType {
  instrument: string;
  skillLevel: SkillLevel;
}

export interface InstructorType {
  userId: string;
  bio?: string;
  instruments?: InstrumentsType[];
  rates?: RatesType;
  placeForLessons?: PlaceForLessonsType;
  studioAddress?: string;
  education?: EducationType[];
  experience?: ExperienceType[];
}

export type IState = InstructorType[];
