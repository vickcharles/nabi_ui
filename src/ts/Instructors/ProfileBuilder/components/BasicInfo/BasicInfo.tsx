import * as React from 'react';
import { Action, Dispatch } from 'redux';
import { ThunkAction } from 'redux-thunk';
import { connect } from 'react-redux';

import { UserState } from '../../../../Users/model';
import { updateUser } from '../../../../Users';
import { InstructorState } from '../../../model';
import { updateInstructor } from '../../../';
import NameLocationBio from './NameLocationBio/NameLocationBio';
import ImageUploader from './ImageUploader';

interface BasicInfoDispatchProps {
  updateInstructor: (instructor: InstructorState) => void;
  updateUser: (user: UserState) => void;
}

interface BasicInfoOwnProps { 
  user: UserState;
  changeAvatar: (id: string, avatar: string) => void;
}

interface BasicInfoState {
  bio: string;
  displayName: string;
}

interface BasicInfoProps extends
  BasicInfoDispatchProps,
  BasicInfoOwnProps { }

export class BasicInfo extends React.Component<BasicInfoProps, BasicInfoState> {
  constructor(props: BasicInfoProps) {
    super(props);

    this.state = {
      bio: '',
      displayName: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleBlurBio = this.handleBlurBio.bind(this);
    this.updateName = this.updateName.bind(this);
  }

  public componentWillMount(): void {
    this.setState({
      ...this.state,
        displayName: `${this.props.user.firstName} ${this.props.user.lastName}`
    });
  }

  public handleChange(event: any): void {
    const target = event.target;
    const value = target.value;
    const name = target.name;
    
    this.setState({
      ...this.state,
      [name]: value
    });
  }

  public handleBlurBio(event: any): void {
    const instructor: InstructorState =  {
      userId: this.props.user.id,
      bio: this.state.bio
    };
    this.props.updateInstructor(instructor);
  }

  public updateName(event: any): void {
    const user =  {
      ...this.props.user,
      displayName: this.state.displayName
    };
    this.props.updateUser(user);
  }

  public render(): JSX.Element {
    return (
      <div>
        <ImageUploader 
          imageChanged={(avatar: string) => {this.props.changeAvatar(this.props.user.id, avatar); }}
        />
        <NameLocationBio 
          user={this.props.user}
          bioBlur={this.handleBlurBio}
          bioChange={this.handleChange}
          updateName={this.updateName}
          handleChange={this.handleChange}
          displayName={this.state.displayName}
        />
      </div>
    );
  }
}

function mapDispatchToProps(
  dispatch: Dispatch<Action | ThunkAction<{}, {}, {}>>,
  _ownProps: BasicInfoOwnProps
): BasicInfoDispatchProps {
  return {
    updateInstructor: (instructor: InstructorState) => dispatch(updateInstructor(instructor)),
    updateUser: (user: UserState) => dispatch(updateUser(user))
  };
}

export default connect(null, mapDispatchToProps)(BasicInfo);
