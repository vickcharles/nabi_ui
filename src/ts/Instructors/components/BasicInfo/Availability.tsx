import * as React from 'react';
import Typography from 'material-ui/Typography';
import GridList, { GridListTile, GridListTileBar } from 'material-ui/GridList';

interface AvailabilityState {
  monEarlyMorning: boolean;
  monLateMorning: boolean;
  monEarlyAfternoon: boolean;
  monLateAfternoon: boolean;
  tueEarlyMorning: boolean;
  tueLateMorning: boolean;
  tueEarlyAfternoon: boolean;
  tueLateAfternoon: boolean;
  wedEarlyMorning: boolean;
  wedLateMorning: boolean;
  wedEarlyAfternoon: boolean;
  wedLateAfternoon: boolean;
  thuEarlyMorning: boolean;
  thuLateMorning: boolean;
  thuEarlyAfternoon: boolean;
  thuLateAfternoon: boolean;
  friEarlyMorning: boolean;
  friLateMorning: boolean;
  friEarlyAfternoon: boolean;
  friLateAfternoon: boolean;
  satEarlyMorning: boolean;
  satLateMorning: boolean;
  satEarlyAfternoon: boolean;
  satLateAfternoon: boolean;
  sunEarlyMorning: boolean;
  sunLateMorning: boolean;
  sunEarlyAfternoon: boolean;
  sunLateAfternoon: boolean;
}

export class Availability extends React.Component<{}, AvailabilityState & any> {
  constructor(props: {}) {
    super(props);

    this.state = {
      monEarlyMorning: false,
      monLateMorning: false,
      monEarlyAfternoon: false,
      monLateAfternoon: false,
      tueEarlyMorning: false,
      tueLateMorning: false,
      tueEarlyAfternoon: false,
      tueLateAfternoon: false,
      wedEarlyMorning: false,
      wedLateMorning: false,
      wedEarlyAfternoon: false,
      wedLateAfternoon: false,
      thuEarlyMorning: false,
      thuLateMorning: false,
      thuEarlyAfternoon: false,
      thuLateAfternoon: false,
      friEarlyMorning: false,
      friLateMorning: false,
      friEarlyAfternoon: false,
      friLateAfternoon: false,
      satEarlyMorning: false,
      satLateMorning: false,
      satEarlyAfternoon: false,
      satLateAfternoon: false,
      sunEarlyMorning: false,
      sunLateMorning: false,
      sunEarlyAfternoon: false,
      sunLateAfternoon: false,
    };
    this.toggleAvailability = this.toggleAvailability.bind(this);
  }

  public toggleAvailability(dayTime: string): void {
    const isAvailable = this.state[dayTime] === false ? true : false;
    this.setState({ [dayTime]: isAvailable });
    console.log(dayTime);
  }

  public render(): JSX.Element {      
    return (
      <div>
        <Typography className="nabi-margin-top-small nabi-margin-bottom-xsmall" variant="body2">
            Availability
          </Typography>
          <Typography >Click to select your availability</Typography>
        <div 
          className="nabi-availability-legend-wrapper nabi-text-center 
          nabi-margin-bottom-small"
        >
          <GridList cellHeight={50} cols={2}>
            <GridListTile>
              <GridListTileBar title="Unavailable" />
            </GridListTile>
            <GridListTile className="nabi-grid-background-nabi">
              <GridListTileBar title="Available"/>
            </GridListTile>
          </GridList>
        </div>

        <div>
          <GridList cellHeight={50} cols={9}>
            <GridListTile key={2} cols={2} className="nabi-background-white nabi-align-center">
              <GridListTileBar className="nabi-text-dark nabi-font-weight-bold" title="Schedule"/>
            </GridListTile>
            <GridListTile className="nabi-align-center">
              <GridListTileBar
                title="M"
              />
            </GridListTile>
            <GridListTile className="nabi-align-center">
              <GridListTileBar
                title="Tu"
              />
            </GridListTile>
            <GridListTile className="nabi-align-center">
              <GridListTileBar
                title="W"
              />
            </GridListTile>
            <GridListTile className="nabi-align-center">
              <GridListTileBar
                title="Th"
              />
            </GridListTile>
            <GridListTile className="nabi-align-center">
              <GridListTileBar
                title="F"
              />
            </GridListTile>
            <GridListTile className="nabi-align-center">
              <GridListTileBar
                title="Sa"
              />
            </GridListTile>
            <GridListTile className="nabi-align-center">
              <GridListTileBar
                title="Su"
              />
            </GridListTile>
            <GridListTile cols={2} className="nabi-align-center">
              <GridListTileBar
                title="Early Morning"
              />
            </GridListTile>
            <GridListTile 
              className={this.state.monEarlyMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('monEarlyMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.tueEarlyMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('tueEarlyMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.wedEarlyMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('wedEarlyMorning')} 
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.thuEarlyMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('thuEarlyMorning')} 
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.friEarlyMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('friEarlyMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.satEarlyMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('satEarlyMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.sunEarlyMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('sunEarlyMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile cols={2} className="nabi-align-center">
              <GridListTileBar
                title="Late Morning"
              />
            </GridListTile>
            <GridListTile 
              className={this.state.monLateMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('monLateMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.tueLateMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('tueLateMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.wedLateMorning ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('wedLateMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.thuLateMorning ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('thuLateMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.friLateMorning ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('friLateMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.satLateMorning ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('satLateMorning')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.sunLateMorning ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('sunLateMorning')}
            >
              <GridListTileBar/>
            </GridListTile>

            <GridListTile cols={2} className="nabi-align-center">
              <GridListTileBar
                title="Early Afternoon"
              />
            </GridListTile>
            <GridListTile 
              className={this.state.monEarlyAfternoon ? 
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('monEarlyAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.tueEarlyAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('tueEarlyAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.wedEarlyAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('wedEarlyAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.thuEarlyAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('thuEarlyAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.friEarlyAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('friEarlyAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.satEarlyAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('satEarlyAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.sunEarlyAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('sunEarlyAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>

            <GridListTile cols={2} className="nabi-align-center">
              <GridListTileBar
                title="Late Afternoon"
              />
            </GridListTile>
            <GridListTile 
              className={this.state.monLateAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('monLateAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.tueLateAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('tueLateAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.wedLateAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('wedLateAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.thuLateAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'} 
              onClick={() => this.toggleAvailability('thuLateAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.friLateAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('friLateAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.satLateAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('satLateAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
            <GridListTile 
              className={this.state.sunLateAfternoon ?
                'nabi-cursor-pointer nabi-grid-background-nabi' : 'nabi-cursor-pointer'}  
              onClick={() => this.toggleAvailability('sunLateAfternoon')}
            >
              <GridListTileBar/>
            </GridListTile>
          </GridList>
        </div>
      </div>
    );
  }
}

export default Availability;
