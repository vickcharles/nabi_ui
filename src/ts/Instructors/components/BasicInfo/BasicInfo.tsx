import * as React from 'react';
import { Action, Dispatch } from 'redux';
import { ThunkAction } from 'redux-thunk';
import { connect } from 'react-redux';

import Typography from 'material-ui/Typography';

import { UserState } from '../../../Users/model';
import { updateUser } from '../../../Users';
import { InstructorState, InstrumentsType, SkillLevel } from '../../model';
import { updateInstructor } from '../../';
import NameLocationBio from './NameLocationBio/NameLocationBio';
import ImageUploader from './ImageUploader';
import Instruments from './Instruments/Instruments';
import SelectedInstrument from './Instruments/SelectedInstrument';
import Rates from './Rates';

interface BasicInfoDispatchProps {
  updateInstructor: (instructor: InstructorState) => void;
  updateUser: (user: UserState) => void;
}

interface BasicInfoOwnProps { 
  user: UserState;
  changeAvatar: (id: string, avatar: string) => void;
}

interface BasicInfoState {
  bio: string;
  displayName: string;
  instrument: string;
  skillLevel: SkillLevel;
  instruments: InstrumentsType[];
}

interface BasicInfoProps extends
  BasicInfoDispatchProps,
  BasicInfoOwnProps { }

export class BasicInfo extends React.Component<BasicInfoProps, BasicInfoState> {
  constructor(props: BasicInfoProps) {
    super(props);

    this.state = {
      bio: '',
      displayName: '',
      instrument: '',
      skillLevel: SkillLevel.beginner,
      instruments: []
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleBlurBio = this.handleBlurBio.bind(this);
    this.updateName = this.updateName.bind(this);
    this.addInstrument = this.addInstrument.bind(this);
    this.deleteInstrument = this.deleteInstrument.bind(this);
  }

  public componentWillMount(): void {
    this.setState({
      ...this.state,
        displayName: `${this.props.user.firstName} ${this.props.user.lastName}`
    });
  }

  public handleChange(event: any): void {
    const target = event.target;
    const value = target.value;
    const name = target.name;
    
    this.setState({
      ...this.state,
      [name]: value
    });
  }

  public handleBlurBio(event: any): void {
    const instructor: InstructorState =  {
      userId: this.props.user.id,
      bio: this.state.bio
    };
    this.updateInstructorCall(instructor);
  }

  public updateName(event: any): void {
    const user =  {
      ...this.props.user,
      displayName: this.state.displayName
    };
    this.props.updateUser(user);
  }

  public addInstrument(event: any): void {
    const instrumentToAdd: InstrumentsType = {
      instrument: this.state.instrument,
      skillLevel: this.state.skillLevel
    };

    if (instrumentToAdd.instrument && instrumentToAdd.skillLevel) {
      this.setState(
        { instruments: [...this.state.instruments, instrumentToAdd] },
        () => {
        const instructor: InstructorState =  {
          userId: this.props.user.id,
          instruments: this.state.instruments
        };
        this.updateInstructorCall(instructor);
      });
    }
  }

  public updateInstructorCall(instructor: InstructorState): void {
    this.props.updateInstructor(instructor);
  }

  public deleteInstrument(instrumentName: string): void {
    this.setState(
      { instruments: this.state.instruments.filter(
        instrument => instrumentName.indexOf(instrument.instrument) === -1)},
      () => {
      const instructor: InstructorState =  {
        userId: this.props.user.id,
        instruments: this.state.instruments
      };
      this.updateInstructorCall(instructor);
    });
  }

  public render(): JSX.Element {
    const selectedInstruments = this.state.instruments.map((instrument, i) => (
      <SelectedInstrument
        key={i}
        instrument={instrument.instrument}
        skillLevel={instrument.skillLevel}
        deleteInstrument={(instrumentName: string) => this.deleteInstrument(instrumentName)}
      />
    ));

    return (
      <div>
        <ImageUploader 
          imageChanged={(avatar: string) => {this.props.changeAvatar(this.props.user.id, avatar); }}
        />
        <NameLocationBio 
          user={this.props.user}
          bioBlur={this.handleBlurBio}
          bioChange={this.handleChange}
          updateName={this.updateName}
          handleChange={this.handleChange}
          displayName={this.state.displayName}
        />
        <Typography className="nabi-margin-top-small" variant="body2">
          Instruments
        </Typography>
        {selectedInstruments}
        <Instruments
          instrument={this.state.instrument}
          skillLevel={this.state.skillLevel}
          handleChange={this.handleChange}
          addInstrument={this.addInstrument}
        />
        <Rates handleChange={this.handleChange} />
      </div>
    );
  }
}

function mapDispatchToProps(
  dispatch: Dispatch<Action | ThunkAction<{}, {}, {}>>,
  _ownProps: BasicInfoOwnProps
): BasicInfoDispatchProps {
  return {
    updateInstructor: (instructor: InstructorState) => dispatch(updateInstructor(instructor)),
    updateUser: (user: UserState) => dispatch(updateUser(user))
  };
}

export default connect(null, mapDispatchToProps)(BasicInfo);
