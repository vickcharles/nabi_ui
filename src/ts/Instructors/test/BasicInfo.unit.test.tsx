import * as React from 'react';
import { shallow } from 'enzyme';
import { BasicInfo } from '../components/BasicInfo/BasicInfo';
import { Role, UserState } from '../../Users/model';

describe('BasicInfo', () => {
  let wrapper: any;
  const updateInstructor = jest.fn();
  const updateInstructorCall = jest.spyOn(BasicInfo.prototype, 'updateInstructorCall');
  const updateUser = jest.fn();
  const changeAvatar: (email: string, avatar: string) => {} = jest.fn();

  const mockUser: UserState = {
    id: '',
    firstName: 'foo',
    lastName: 'bar',
    email: 'oreo',
    password: '',
    zipCode: '',
    role: Role.instructor,
    hearAboutUs: '',
    displayName: ''
  };

  beforeEach(() => {
    wrapper = shallow(
      <BasicInfo 
        user={mockUser}
        changeAvatar={changeAvatar}
        updateInstructor={updateInstructor}
        updateUser={updateUser}
      />
    );
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Method handleChange()', () => {
    function test_handleChange(theName: string, theValue: string): void {
      describe(`When the event contains ${theName} for name and ${theValue} for value`, () => {
        beforeEach(() => {
          const e = {
            target: {
              name: theName,
              value: theValue
            }
          };

          wrapper.instance().handleChange(e);
        });

        it(`Sets the state's ${theName} to ${theValue}`, () => {
          expect(wrapper.state(theName)).toBe(theValue);
        });
      });
    }

    test_handleChange('bio', 'yo naci en esta rivera');
    test_handleChange('displayName', 'La Va');
  });

  describe('Method handleBlurBio()', () => {
    beforeEach(() => {
      wrapper.instance().handleBlurBio();
    });

    it('Calls updateInstructor', () => {
      expect(updateInstructor).toBeCalled();
    });
  });

  describe('Method updateName()', () => {
    beforeEach(() => {
      wrapper.instance().updateName();
    });

    it('Calls updateUser', () => {
      expect(updateUser).toBeCalled();
    });
  });

  describe('Method addInstrument()', () => {
    beforeEach(() => {
      wrapper.instance().addInstrument();
    });

    it('Calls updateInstructorCall', () => {
      expect(updateInstructorCall).toBeCalled();
    });
  });

  describe('Method deleteInstrument()', () => {
    beforeEach(() => {
      wrapper.instance().deleteInstrument();
    });

    it('Calls updateInstructorCall', () => {
      expect(updateInstructorCall).toBeCalled();
    });
  });
});
