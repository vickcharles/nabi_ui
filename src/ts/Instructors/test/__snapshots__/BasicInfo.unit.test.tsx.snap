// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfo Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BasicInfo
    changeAvatar={[MockFunction]}
    updateInstructor={[MockFunction]}
    updateUser={[MockFunction]}
    user={
      Object {
        "displayName": "",
        "email": "oreo",
        "firstName": "foo",
        "hearAboutUs": "",
        "id": "",
        "lastName": "bar",
        "password": "",
        "role": "instructor",
        "zipCode": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ImageUploader
          imageChanged={[Function]}
        />,
        <NameLocationBio
          bioBlur={[Function]}
          bioChange={[Function]}
          displayName="foo bar"
          handleChange={[Function]}
          updateName={[Function]}
          user={
            Object {
              "displayName": "",
              "email": "oreo",
              "firstName": "foo",
              "hearAboutUs": "",
              "id": "",
              "lastName": "bar",
              "password": "",
              "role": "instructor",
              "zipCode": "",
            }
          }
        />,
        <WithStyles(Typography)
          className="nabi-margin-top-small"
          variant="body2"
        >
          Instruments
        </WithStyles(Typography)>,
        Array [],
        <Instruments
          addInstrument={[Function]}
          handleChange={[Function]}
          instrument=""
          skillLevel="Beginner"
        />,
        <Rates
          fortyFiveMinsRate={0}
          handleChange={[Function]}
          ninetyMinsRate={0}
          sixtyMinsRate={0}
          thirtyMinsRate={0}
          updateRates={[Function]}
        />,
        <PlaceForLessons
          handleChange={[Function]}
          handleChangePlaceForLessons={[Function]}
          home={false}
          online={false}
          studio={false}
          studioAddress=""
          updateStudioAddress={[Function]}
        />,
        <Availability
          updateInstructor={[Function]}
          userId=""
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "imageChanged": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bioBlur": [Function],
          "bioChange": [Function],
          "displayName": "foo bar",
          "handleChange": [Function],
          "updateName": [Function],
          "user": Object {
            "displayName": "",
            "email": "oreo",
            "firstName": "foo",
            "hearAboutUs": "",
            "id": "",
            "lastName": "bar",
            "password": "",
            "role": "instructor",
            "zipCode": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Instruments",
          "className": "nabi-margin-top-small",
          "variant": "body2",
        },
        "ref": null,
        "rendered": "Instruments",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addInstrument": [Function],
          "handleChange": [Function],
          "instrument": "",
          "skillLevel": "Beginner",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fortyFiveMinsRate": 0,
          "handleChange": [Function],
          "ninetyMinsRate": 0,
          "sixtyMinsRate": 0,
          "thirtyMinsRate": 0,
          "updateRates": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleChange": [Function],
          "handleChangePlaceForLessons": [Function],
          "home": false,
          "online": false,
          "studio": false,
          "studioAddress": "",
          "updateStudioAddress": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "updateInstructor": [Function],
          "userId": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ImageUploader
            imageChanged={[Function]}
          />,
          <NameLocationBio
            bioBlur={[Function]}
            bioChange={[Function]}
            displayName="foo bar"
            handleChange={[Function]}
            updateName={[Function]}
            user={
              Object {
                "displayName": "",
                "email": "oreo",
                "firstName": "foo",
                "hearAboutUs": "",
                "id": "",
                "lastName": "bar",
                "password": "",
                "role": "instructor",
                "zipCode": "",
              }
            }
          />,
          <WithStyles(Typography)
            className="nabi-margin-top-small"
            variant="body2"
          >
            Instruments
          </WithStyles(Typography)>,
          Array [],
          <Instruments
            addInstrument={[Function]}
            handleChange={[Function]}
            instrument=""
            skillLevel="Beginner"
          />,
          <Rates
            fortyFiveMinsRate={0}
            handleChange={[Function]}
            ninetyMinsRate={0}
            sixtyMinsRate={0}
            thirtyMinsRate={0}
            updateRates={[Function]}
          />,
          <PlaceForLessons
            handleChange={[Function]}
            handleChangePlaceForLessons={[Function]}
            home={false}
            online={false}
            studio={false}
            studioAddress=""
            updateStudioAddress={[Function]}
          />,
          <Availability
            updateInstructor={[Function]}
            userId=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "imageChanged": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bioBlur": [Function],
            "bioChange": [Function],
            "displayName": "foo bar",
            "handleChange": [Function],
            "updateName": [Function],
            "user": Object {
              "displayName": "",
              "email": "oreo",
              "firstName": "foo",
              "hearAboutUs": "",
              "id": "",
              "lastName": "bar",
              "password": "",
              "role": "instructor",
              "zipCode": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Instruments",
            "className": "nabi-margin-top-small",
            "variant": "body2",
          },
          "ref": null,
          "rendered": "Instruments",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addInstrument": [Function],
            "handleChange": [Function],
            "instrument": "",
            "skillLevel": "Beginner",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fortyFiveMinsRate": 0,
            "handleChange": [Function],
            "ninetyMinsRate": 0,
            "sixtyMinsRate": 0,
            "thirtyMinsRate": 0,
            "updateRates": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": [Function],
            "handleChangePlaceForLessons": [Function],
            "home": false,
            "online": false,
            "studio": false,
            "studioAddress": "",
            "updateStudioAddress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "updateInstructor": [Function],
            "userId": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
